flight/targets/omnibusf3/board-info/board_hw_defs.c

```diff
@@ -0,0 +1,977 @@
+	},
+	.force_1MHz = true,
+	.channels = pios_tim_servoport_pins,
+	.num_channels = NELEMENTS(pios_tim_servoport_pins),
+};
+
+#endif	/* PIOS_INCLUDE_SERVO && PIOS_INCLUDE_TIM */
+
+
+
+/*
+ * PPM Input
+ */
+#if defined(PIOS_INCLUDE_PPM)
+#include <pios_pwm_priv.h>
+
+/*
+ * 	INPUT - PPM
+ *
+ * 	TIM3 CH1 PB4
+ */
+static const struct pios_tim_channel pios_tim_rxport_ppm[] = {
+	{
+		.timer = TIM3,
+		.timer_chan = TIM_Channel_1,
+		.remap = GPIO_AF_2,
+		.pin = {
+			.gpio = GPIOB,
+			.init = {
+				.GPIO_Pin = GPIO_Pin_4,
+				.GPIO_Speed = GPIO_Speed_2MHz,
+				.GPIO_Mode  = GPIO_Mode_AF,
+				.GPIO_OType = GPIO_OType_PP,
+				.GPIO_PuPd  = GPIO_PuPd_UP
+			},
+			.pin_source = GPIO_PinSource4,
+		},
+	},
+};
+#endif
+
+/*
+ * PPM Input
+ */
+#if defined(PIOS_INCLUDE_PPM)
+#include <pios_ppm_priv.h>
+static const struct pios_ppm_cfg pios_ppm_cfg = {
+	.tim_ic_init = {
+		.TIM_ICPolarity = TIM_ICPolarity_Rising,
+		.TIM_ICSelection = TIM_ICSelection_DirectTI,
+		.TIM_ICPrescaler = TIM_ICPSC_DIV1,
+		.TIM_ICFilter = 0x0,
+		.TIM_Channel = TIM_Channel_1,
+	},
+	/* Use only the first channel for ppm */
+	.channels = pios_tim_rxport_ppm,
+	.num_channels = 1,
+};
+
+#endif //PPM
+
+#if defined(PIOS_INCLUDE_ADC)
+#include "pios_adc_priv.h"
+#include "pios_internal_adc_priv.h"
+
+//-------------------------
+// ADC
+// ADC0 : PA0 ADC1_IN1  VBat
+// ADC1 : PA1 ADC1_IN2  Current
+// ADC2 : PB1 ADC3_IN1 RSSI
+//-------------------------
+
+static const struct pios_internal_adc_cfg internal_adc1_cfg = {
+	.dma = {
+		.irq = {
+			.flags   = (DMA1_FLAG_TC1 | DMA1_FLAG_TE1 | DMA1_FLAG_HT1 | DMA1_FLAG_GL1),
+			.init    = {
+				.NVIC_IRQChannel                   = DMA1_Channel1_IRQn,
+				.NVIC_IRQChannelPreemptionPriority = PIOS_IRQ_PRIO_HIGH,
+				.NVIC_IRQChannelSubPriority        = 0,
+				.NVIC_IRQChannelCmd                = ENABLE,
+			},
+		},
+		.rx = {
+			.channel = DMA1_Channel1,
+			.init    = {
+				.DMA_Priority           = DMA_Priority_High,
+			},
+		}
+	},
+	.half_flag = DMA1_IT_HT1,
+	.full_flag = DMA1_IT_TC1,
+	.oversampling = 32,
+	.adc_pin_count = 2,
+	.adc_pins = {
+		{GPIOA, GPIO_Pin_0, ADC_Channel_1, true},  // Voltage
+		{GPIOA, GPIO_Pin_1, ADC_Channel_2, true},  // Current
+	},
+	.adc_dev_master = ADC1,
+};
+#endif /* PIOS_INCLUDE_ADC */
+
+
+#if defined(PIOS_INCLUDE_GCSRCVR)
+#include "pios_gcsrcvr_priv.h"
+#endif	/* PIOS_INCLUDE_GCSRCVR */
+
+
+#if defined(PIOS_INCLUDE_RCVR)
+#include "pios_rcvr_priv.h"
+#endif /* PIOS_INCLUDE_RCVR */
+
+
+#if defined(PIOS_INCLUDE_USB)
+#include "pios_usb_priv.h"
+
+static const struct pios_usb_cfg pios_usb_main_cfg = {
+	.irq = {
+		.init    = {
+			.NVIC_IRQChannel                   = USB_LP_CAN1_RX0_IRQn,
+			.NVIC_IRQChannelPreemptionPriority = PIOS_IRQ_PRIO_LOW,
+			.NVIC_IRQChannelSubPriority        = 0,
+			.NVIC_IRQChannelCmd                = ENABLE,
+		},
+	},
+#ifdef BOOTLOADER_VERSION
+	// Only use vsense in bootloader, since PB5 is also the first output
+	// channel.
```

Comment:
LOL

Reply:
Not too bad-- seppuku uses the same trick